flink-yarn/src/main/java/org/apache/flink/yarn/YarnTaskManagerRunner.java

```diff
@@ -75,34 +84,47 @@ public static void runYarnTaskManager(String[] args, final Class<? extends YarnT
 				"specified in the Flink config: " + flinkTempDirs);
 		}
 
-		LOG.info("YARN daemon runs as '" + UserGroupInformation.getCurrentUser().getShortUserName() +
-			"' setting user to execute Flink TaskManager to '" + yarnClientUsername + "'");
-
 		// tell akka to die in case of an error
 		configuration.setBoolean(ConfigConstants.AKKA_JVM_EXIT_ON_FATAL_ERROR, true);
 
-		UserGroupInformation ugi = UserGroupInformation.createRemoteUser(yarnClientUsername);
-		for (Token<? extends TokenIdentifier> toks : UserGroupInformation.getCurrentUser().getTokens()) {
-			ugi.addToken(toks);
+		String keytabPath = null;
+		if(remoteKeytabPath != null) {
+			File f = new File(currDir, ConfigConstants.KEYTAB_FILE_NAME);
```

Comment:
The name is not configurable (user provided) but we use a constant value. Is there any reason to keep the name unique?